docs/code-docs/source/zero3.rst

```diff
@@ -293,3 +293,38 @@ parallelism to fit them in limited GPU memory.
 
 .. autoclass:: deepspeed.zero.TiledLinear
     :members:
+
+
+Debugging
+---------
+
+Debugging ZeRO training is complicated by the partitioning of parameters, gradients, and optimizer states.
+None of these 3 groups of parameters can be normally accessed because of that. To overcome that DeepSpeed provides the following utilities for accessing individual model states in their unpartitioned form.
+
+Please note that routines utilities must be called by all processes, even if you decide to do something with the result only in the main process. If all processes don't participate these utilities will hang waiting for all processes to send their contribution.
```

Comment:
```suggestion
Debugging ZeRO training is complicated by the partitioning of parameters, gradients, and optimizer states.
None of these 3 groups of tensors (model states) can be normally accessed because of that. To overcome that DeepSpeed provides the following routines for accessing individual model states in their unpartitioned form.

Important: Please note that these utilities must be called by all processes participating in the training, even if you decide to do something with the result only in the main process. If all processes don't participate these utilities will hang waiting for all processes to send their contribution.

Additionally, you must be aware that these routines return correct data only in specific phases of the training. So for examples the gradients are valid after ``backward`` and before ``step``. The optimizer states are updated after ``step``. Same goes for fp32 master weights.
```

Reply:
sorry, I was still editing the suggestion - I reloaded it - now it's good to go.